source/rst/continuous_time/seir_model_sde.rst

```diff
@@ -0,0 +1,913 @@
+
+Introduction to SDEs: Aggregate Shocks to Transmission Rates
+==============================================================
+
+We will start by extending our model to include randomness in :math:`R(t)`, which makes it a system of Stochastic Differential Equations (SDEs).
+
+Continuous-Time Stochastic Processes
+-----------------------------------
+
+In continuous-time, there is an important distinction between randomness that leads to continuous paths vs. those which may have jumps (which are almost surely right-continuous).  The most tractable of these includes the theory of `Levy Processes <https://en.wikipedia.org/wiki/L%C3%A9vy_process>`_.
+
+**TBD:** Add definition of levy processes and the intuitive connection between stationary increments and independence of increments.  Also, is there intuition why additive processes in continuous time lead to additive SDEs/etc.?  Might be useful.
+
+Among the appealing features of Levy Processes is that they fit well into the sorts of Markov modeling techniques that economists tend to use in discrete time...
+
+Unlike in discrete-time, where a modeller has license to be creative, the rules of continuous-time stochastic processes are much more strict.  In practice, there are only two types of Levy Processes that can be used without careful measure theory.
+
+#. `Weiner Processes <https://en.wikipedia.org/wiki/Wiener_process>`__ (as known as Brownian Motion) which leads to a diffusion equations, and is the only continuous-time Levy process with continuous paths
+#. `Poisson Processes <https://en.wikipedia.org/wiki/Poisson_point_process>`__ with an arrival rate of jumps in the variable.
+
+Every other Levy Process you will typically work with is composed of those building blocks (e.g. a `Diffusion Process <https://en.wikipedia.org/wiki/Diffusion_process>`__ such as Geometric Brownian Motion is a transformation of a Weiner process, and a `jump diffusion <https://en.wikipedia.org/wiki/Jump_diffusion#In_economics_and_finance>`__ is a diffusion process with a Poisson arrival of jumps).
+
+In this section, we will look at example of a diffusion process.
+
+
+Shocks to Transmission Rates
+------------------------------
+
+Consider that the effective transmission rate :math:`R(t)` could depend on degrees of randomness in behavior and implementation.  For example,
+
+* Misinformation on Facebook spreading non-uniformly
+* Large political rallies, elections, or protests
+* Deviations in the implementation and timing of lockdown policy within demographics, locations, or businesses within the system.
+* Aggregate shocks in opening/closing industries
+
+To implement this, we will add on a diffusion term to :eq:`Rode` with an instantaneous volatility of :math:`\zeta \sqrt{R}`.  The scaling by the :math:`\sqrt{R}` ensures that the process can never go negative since the variance converges to zero as R goes to zero.
```

Comment:
Not sure it actually ends up as a provably positive process. At least the argument and intuition aren't necessarily correct. `u' = -sqrt(abs(u))` is an example where the rate goes to zero but this process hits exactly zero in finite time, at which point its solution is non-unique but it's either 0 or it goes negative. I think the fact that the diffusion length scales by another sqrt might make this work out, but it's not exactly trivial.

Reply:
I added in the scaling by the sqrt when we discussed this previously, and you had suggested it was a good way to keep things positive.

But if I have learned anythign over the years, it is that continuous time stochastic processes are is often counterintutive, and that the details matter.

For this one, I think the finance crew calls it the CIR process: https://en.wikipedia.org/wiki/Cox%E2%80%93Ingersoll%E2%80%93Ross_model

It says there that the scaling avoids the possibility of negative interest rates for all `a` and `b`, and an absorbing state of zero is precluded as long as the drift is strong enough relative to the variance.  I think this comes out of the ergodic distribution being Gamma for the right combination of parameters: https://en.wikipedia.org/wiki/Cox%E2%80%93Ingersoll%E2%80%93Ross_model#Distribution

With that said, I don't think it means that the SDE simulators would necessarily always keep things positive?  Which might be what I ran into later in the code (which you might see in https://github.com/QuantEcon/lecture-source-jl/blob/459e806e3acaa3371013112ed10eae7c5f38ce4a/source/rst/continuous_time/seir_model_sde.rst#implementing-the-complete-system  where I fudge the variance to avoid a sqrt of a negative.  Probably better ways to do this.

I think a little adhoc on the SDE side for this is OK as long as we are clea rwhat is going on.

Reply:
>It says there that the scaling avoids the possibility of negative interest rates for all a and b, and an absorbing state of zero is precluded as long as the drift is strong enough relative to the variance.

Alright, cool. I think we should mention that the explanation given is really just a heuristic, and to know it's going to be positive requires more details. 

>With that said, I don't think it means that the SDE simulators would necessarily always keep things positive?

Yeah, being that close to zero can cause numerical difficulties

Reply:
Comment added